migrations/00002_custom_user_fields.sql

```diff
@@ -0,0 +1,5 @@
+-- start a transaction
+BEGIN;
+ALTER TABLE auth.users
+ADD COLUMN custom JSONB;
```

Comment:
Any specific reason to call this `custom` instead of `metadata`?

Reply:
Naming things is always tricky, I'm fine with `metadata`, no strong opinion on the matter - `metadata` then?

Reply:
Yea it's a good question.

I see this column more as a way for developers to attach data during signup rather than storing general metadata for users. In fact, I'm not sure we should encourage developers to use this unstructured JSON `metadata` column to store user metadata. User metadata should probably be stored in a `public.profiles` table.

Supabase has a `metadata` column on their `auth.users` table but they mention this in [their docs](https://supabase.com/docs/reference/javascript/auth-update):

> User metadata: It's generally better to store user data in a table inside your public schema (i.e. public.users). Use the update() method if you have data which rarely changes or is specific only to the logged in user.

The naming of this column could have an impact on the way developers use the column and the way we document the recommended usage of the column.

Reply:
@nunopato Any thoughts on this?

Reply:
I think the `metadata` column makes sense for data that does not change often, consists of a single value, and is known at signup time (e.g., `family_name`, `age`, etc). Having a separate table makes sense for data like a user's address where you would have to store multiple values (e.g., street, address, etc). I think this is how we should document its usage.

Reply:
Let's change to `metadata` then!